### ISSUE TITLE
vmdisplay: texture is not deleted explicitly?

### ISSUE THREAD

**Issue by xinyunliu** [intel/ias#58]
The code was trying to free the unused texture in clear_rec(), but  the hbuflist not correctly updated in update_hyper_dmabuf_list(): will not update the hbuflist  when `create_new_hyper_dmabuf_buffer()`.  So the texture has leackage?

```
static void clear_rec(struct buffer_list *l, int i)
{
        l->l[i].age = 0;
        glDeleteTextures(2, l->l[i].textureId);
        if (l->l[i].buffer)
                wl_buffer_destroy(l->l[i].buffer);
        l->l[i].textureId[0] = 0;
        l->l[i].textureId[1] = 0;
}
```

`find_rec()` normally return -1, and so will call create_new_hyper_dmabuf_buffer() directly.

```
static void update_hyper_dmabuf_list(int id, int old_id)
{
        int r = find_rec(&hyper_dmabuf_list, id);
        age_list(&hyper_dmabuf_list);

        if (r >= 0) {
                if (old_id == 0 ||
                    hyper_dmabuf_list.l[r].width != surf_width ||
                    hyper_dmabuf_list.l[r].height != surf_height) {
                        clear_rec(&hyper_dmabuf_list, r);
                        create_new_hyper_dmabuf_buffer();
                } else {
                        last_rec(&hyper_dmabuf_list, r);
                        current_textureId[0] =
                            hyper_dmabuf_list.l[r].textureId[0];
                        current_textureId[1] =
                            hyper_dmabuf_list.l[r].textureId[1];
                        current_buffer = hyper_dmabuf_list.l[r].buffer;
                }
        } else {
                create_new_hyper_dmabuf_buffer();
        }
}


```

**Comment by xinyunliu** [comment]
@satyeshwar @downor @shuangwan01  Can you have look at the issue? Thanks!

**Comment by downor** [comment]
I think BO for the texture won’t be evicted for a while as it can be reused in the future... but I will double check

On Jun 4, 2019, at 8:05 PM, Liu, Xinyun <notifications@github.com<mailto:notifications@github.com>> wrote:


@satyeshwar<https://github.com/satyeshwar> @downor<https://github.com/downor> @shuangwan01<https://github.com/shuangwan01> Can you have look at the issue? Thanks!

**Comment by downor** [comment]
Xinyun,

Can you re-check this please? Current upstream version works just fine in terms of texture freeing on customer's use-case.

**Comment by xinyunliu** [comment]
> Xinyun,
> 
> Can you re-check this please? Current upstream version works just fine in terms of texture freeing on customer's use-case.

Confirmed. It's fixed in ba6053e34b38a37353a1fa77bd0e3dd29c96448a
